crates/red_knot_python_semantic/resources/mdtest/narrow/conditionals_not_eq.md

```diff
@@ -0,0 +1,56 @@
+# Narrowing for `!=` conditionals
+
+## `x != None`
+
+```py
+x = None if flag else 1
+
+if x != None:
+    reveal_type(x)  # revealed: Literal[1]
+```
+
+## `!=` for other singleton types
+
+```py
+x = True if flag else False
+
+if x != False:
+    reveal_type(x)  # revealed: Literal[True]
+```
+
+## `x != y` where `y` is of literal type
+
+```py
+x = 1 if flag else 2
+
+if x != 1:
+    reveal_type(x)  # revealed: Literal[2]
+```
+
+## `x != y` where `y` is a single-valued type
+
+```py
+class A: ...
+
+
+class B: ...
+
+
+C = A if flag else B
+
+if C != A:
+    reveal_type(C)  # revealed: Literal[B]
+```
+
+## `!=` for non-singleton types
+
+Non-singleton types should *not* narrow the type: two instances of a
+non-singleton class may occupy different addresses in memory even if
+they compare equal.
+
+```py
+x = 1
+
+if x != 0:
+    reveal_type(x)  # revealed: Literal[1]
+```
```

Comment:
> This test doesn't currently seem to make sense

Absolutely. I just pushed the version which I had lying around locally already, but didn't commit+push because I got interrupted (sorry). I think it's more or less equivalent to what you proposed, but let me know if I should add something similar to `a = A(); b = B(); if a != b: ...`.

Reply:
The test you pushed looks sufficient!